### ISSUE TITLE
Contact Form: datepicker is not accessible to keyboard users

### ISSUE THREAD

**Issue by monsieur-z** [Automattic/jetpack#34501]
### Impacted plugin

Jetpack

### Quick summary

Keyboard users can focus on the date field of a _Contact Form_. However, the date picker component itself is not focusable and, therefore, not usable.

Additionally, users can enter any string: there's no validation against any date format. Site owners may want a consistent format for the responses date field.

<img width="500" alt="Screenshot 2023-12-06 at 3 39 07 PM" src="https://github.com/Automattic/jetpack/assets/1620183/5a7b74da-500a-43bf-96e9-db82e5f08430">

### Steps to reproduce

1. Create a new post
2. Add a _Contact Form_ block
3. Add a _Date Picker_ field
4. Open the preview
5. Navigate through the form using the Tab key of the keyboard
6. Notice you can't interact with the date picker itself
7. Enter any string in the field
8. Submit the form
9. Notice there's no error message associated with the date field

### A clear and concise description of what you expected to happen.

Keyboard users should be able to interact with the date picker. At the very least, they should be able to enter a date manually. While they technically can, users might be confused by the lack of date format.

### What actually happened

Users can't interact with the date picker using the keyboard.

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

**Comment by jeherve** [comment]
Related: #18863
